### PR TITLE
REPO-1361: Adds rake task and service object to disable every tenant cache

### PR DESCRIPTION
**Pull request by okeen**

### CODE REVIEW

File: app/models/concerns/hyku_addons/account_behavior.rb [hunk 1]
```diff
@@ -79,9 +79,9 @@ def switch_host!(cname)
       def setup_tenant_cache(is_enabled)
         Rails.application.config.action_controller.perform_caching = is_enabled
         ActionController::Base.perform_caching = is_enabled
-        # rubocop:disable Style/ConditionalAssignment
```
Comment (BertZZ) [inline app/models/concerns/hyku_addons/account_behavior.rb:82]: :pray:

File: app/services/hyku_addons/cache_expiration_service.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        Rails.logger.info "Expiring cache for #{account.cname}"
+        next if account.redis_endpoint.is_a?(NilRedisEndpoint)
+
+        expire_cache_for(account)
+      end
+    end
+
+    def expire_cache_for(account)
+      account.redis_endpoint.switch!
+      account.setup_tenant_cache(true)
```
Comment (BertZZ) [inline app/services/hyku_addons/cache_expiration_service.rb:16]: Could we not call account.switch! and save a line?

File: spec/models/concerns/hyku_addons/account_behavior_spec.rb [hunk 3]
```diff
@@ -43,7 +44,7 @@
       end
     end
 
-    context "when cashe is disabled" do
+    context "when cache is disabled" do
```
Comment (BertZZ) [inline spec/models/concerns/hyku_addons/account_behavior_spec.rb:47]: :scream:

File: spec/services/hyku_addons/cache_expiration_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  end
+
+  describe "perform" do
+    before do
+      allow(service).to receive(:expire_cache_for)
+      service.perform
+    end
+
+    it "updates the frontend_url of the Account" do
+      expect(service).to have_received(:expire_cache_for).once
```
Comment (BertZZ) [inline spec/services/hyku_addons/cache_expiration_service_spec.rb:23]: What does this have to do with the frontend URL?

File: app/services/hyku_addons/cache_expiration_service.rb (hunk 2 above)
Comment (okeen) [reply app/services/hyku_addons/cache_expiration_service.rb:16]: There are some extra things made in `account.switch!` that we really don't need here, including loading the settings which can be costly.
This should avoid that

File: spec/services/hyku_addons/cache_expiration_service_spec.rb (hunk 4 above)
Comment (okeen) [reply spec/services/hyku_addons/cache_expiration_service_spec.rb:23]: Copy pasta thing, good catch

**Review by BertZZ — APPROVED** [review]
